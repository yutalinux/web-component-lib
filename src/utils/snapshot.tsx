import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
