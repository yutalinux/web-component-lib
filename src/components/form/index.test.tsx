import { snapshotElement } from "@/utils/snapshot.test";
import { expect, test } from "vitest";
import { Form } from ".";

test("Form Snapshot", () => {
  expect(snapshotElement(Form, null)).toMatchSnapshot();
});
