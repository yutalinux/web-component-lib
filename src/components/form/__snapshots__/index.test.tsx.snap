// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form Snapshot 1`] = `"<form class="secondary secondary-border p-4 rounded-md flex flex-col gap-4"></form>"`;
